network/p2p/p2p.go

```diff
@@ -89,12 +89,13 @@ type serviceImpl struct {
 	topicsMu deadlock.RWMutex
 }
 
-// AlgorandWsProtocol defines a libp2p protocol name for algorand's websockets messages
-const AlgorandWsProtocol = "/algorand-ws/1.0.0"
+// AlgorandWsProtocolV1 defines a libp2p protocol name for algorand's websockets messages
+// as the very initial release
+const AlgorandWsProtocolV1 = "/algorand-ws/1.0.0"
 
-// algorandGUIDProtocolPrefix defines a libp2p protocol name for algorand node telemetry GUID exchange
-const algorandGUIDProtocolPrefix = "/algorand-telemetry/1.0.0/"
-const algorandGUIDProtocolTemplate = algorandGUIDProtocolPrefix + "%s/%s"
+// AlgorandWsProtocolV22 defines a libp2p protocol name for algorand's websockets messages
+// as a version supporting peer metadata and wsnet v2.2 protocol features
+const AlgorandWsProtocolV22 = "/algorand-ws/2.2.0"
```

Comment:
Q: should it somehow in sync with `wsnetwork.ProtocolVersion`?

Reply:
Ideally one would reference the other..

Reply:
a "problem" with this approach is wsnet v1.0 is not supported but we still need `/algorand-ws/1.0.0` in p2p. I suggest to unify 2.2 in later release after the consensus upgrade